src/index.ts

```diff
@@ -35,20 +35,21 @@ export const runTopologyWithNats =
     const { unpack, loadSnapshot, persistSnapshot } = fns
     const data = unpack(msg.data)
     const numAttempts = msg.info.redeliveryCount
-    const streamData = getStreamDataFromMsg(msg)
     const isRedelivery = numAttempts > 1
+    debug('Redelivery %s', isRedelivery)
     const { emitter, promise, getSnapshot } = isRedelivery
       ? // Resume topology based on unique stream data
-        resumeTopology(spec, await loadSnapshot(streamData))
+        resumeTopology(spec, await loadSnapshot(msg))
       : // Run topoloty with data from msg
-        runTopology(spec, dag, { data, meta: streamData })
+        runTopology(spec, dag, { data })
+    const streamData = getStreamDataFromMsg(msg)
     const persist = (snapshot: Snapshot) => {
       const streamSnapshot = { ...snapshot, ...streamData, numAttempts }
```

Comment:
So you decided to leave `streamData`?

Reply:
Yeah. It was annoying to have to add that data. It's useful information even if you don't use it as a uniqueness constraint.